lib-injection/host_inject.rb

```diff
@@ -1,48 +1,113 @@
-# Keep in sync with auto_inject.rb
 return if ENV['DD_TRACE_SKIP_LIB_INJECTION'] == 'true'
 
-require 'rubygems'
-require 'rbconfig'
-
-ruby_api_version = RbConfig::CONFIG['ruby_version']
-
-dd_lib_injection_path = "/opt/datadog/apm/library/ruby/#{ruby_api_version}"
-
-def debug_log(msg)
-  $stdout.puts msg if ENV['DD_TRACE_DEBUG'] == 'true'
-end
-
-debug_log "[datadog] Loading from #{dd_lib_injection_path}..."
-
 begin
+  require 'rubygems'
   require 'open3'
   require 'bundler'
   require 'bundler/cli'
   require 'shellwords'
   require 'fileutils'
+  require 'json'
 
-  support_message = 'For help solving this issue, please contact Datadog support at https://docs.datadoghq.com/help/.'
+  def dd_debug_log(msg)
+    $stdout.puts "[datadog] #{msg}" if ENV['DD_TRACE_DEBUG'] == 'true'
+  end
+
+  def dd_error_log(msg)
+    warn "[datadog] #{msg}"
+  end
+
+  def dd_skip_injection!
+    ENV['DD_TRACE_SKIP_LIB_INJECTION'] = 'true'
+  end
+
+  def dd_send_telemetry(events)
+    pid = Process.respond_to?(:pid) ? Process.pid : 0 # Not available on all platforms
+
+    tracer_version = if File.exist?('/opt/datadog/apm/library/ruby/version.txt')
```

Comment:
This `version.txt` is not familiar to me, where does it come from?

Reply:
During the process for building the gem, we wrote it.
https://github.com/DataDog/dd-trace-rb/blob/master/.gitlab-ci.yml#L94